mongonaut/forms/form_mixins.py

```diff
@@ -78,8 +83,13 @@ def get_form_field_dict(self, model_dict):
         keyed by form field.  Each value is a  keyed 4 tuple of:
         (widget, mode_field_instance, model_field_type, field_key)
         """
-        return_dict = {}
-        for field_key, field_dict in model_dict.iteritems():
+        return_dict = OrderedDict()
+        if hasattr(self.model, 'Meta') and hasattr(self.model.Meta, 'ordering'):
+            ordering = tuple(o for o in self.model.Meta.ordering if o in model_dict.iterkeys())
```

Comment:
Good call. I have looked at the source code both in **Django** and **mongoengine**, I think I've mixed up the usage of Meta class ordering (for queryset). And in fact, the ordering should be preserved from  **metaclass** `__new__`, like a `SortedDict` in **Django** (which itself is going to be deprecated and replaced by OrderedDict). 
I have renamed the `o` to be more descriptive and unless **mongoengine** has fixed the `OrderedDict` issue, this is a workaround to add another attribute in model's Meta class (I call it **form_fields_ordering**). 

And you're right the temporary **tuple** comprehension is there to preserve the ordering and once it's loaded by `fromkeys` the duplicate will be gone. And tuple will be more efficient and backward compatible.